src/main/java/com/xjeffrose/chicago/ChicagoMessageAggregator.java

```diff
@@ -0,0 +1,37 @@
+package com.xjeffrose.chicago;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageDecoder;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ChicagoMessageAggregator extends MessageToMessageDecoder<ChicagoMessage> {
+  private final AtomicBoolean started = new AtomicBoolean();
+  private AggregatedChicagoMessage aggregatedMessage;
+
+  private void aggregate(ChicagoMessage msg) {
+    if (started.get()) {
+      aggregatedMessage.appendVal(msg.getVal());
+    } else {
+      aggregatedMessage = new AggregatedChicagoMessage(msg);
+      started.set(true);
+    }
+  }
+
+
+  @Override
+  protected void decode(ChannelHandlerContext ctx, ChicagoMessage msg, List<Object> out) throws Exception {
+
+    switch (msg.getOp()) {
+      case STREAM_RESPONSE:
+        aggregate(msg);
+        if (new String(msg.getVal()).contains(ChiUtil.delimiter)) {
+          out.add(aggregatedMessage);
+        }
+        break;
+      default:
+        out.add(msg);
```

Comment:
I think you need to call `ReferenceCountUtil.retain(msg)` here to compensate for the release that will happen after you return. 

See https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/MessageToMessageDecoder.java#L90